src/chi-vue/src/components/data-table/DataTableUtils.ts

```diff
@@ -0,0 +1,18 @@
+export function printDataTable(content: string, title: string) {
+  const win = window.open();
+  const styleSheets =
+    Array.from(document.head.getElementsByTagName('link'))
+      .map(el => el.outerHTML)
+      .reduce((previousValue, currentValue) => previousValue + currentValue, '') +
+    Array.from(document.head.getElementsByTagName('style'))
+      .map(el => el.outerHTML)
+      .reduce((previousValue, currentValue) => previousValue || '' + currentValue, '');
+
+  win?.document.write(`<head><title>${title}</title>${styleSheets}</head><body class='chi'>${content}</body>`);
+  win?.document.close();
+
+  win?.setTimeout(() => {
+    win.print();
+    win.close();
+  }, 0);
```

Comment:
```suggestion

  if (!win) {
    throw new Error("Could not open print window. ");
  }
  
  const styleSheets =
    Array.from(document.head.getElementsByTagName('link'))
      .map(el => el.outerHTML)
      .reduce((previousValue, currentValue) => previousValue + currentValue, '') +
    Array.from(document.head.getElementsByTagName('style'))
      .map(el => el.outerHTML)
      .reduce((previousValue, currentValue) => previousValue || '' + currentValue, '');

  win.document.write(`<head><title>${title}</title>${styleSheets}</head><body class='chi'>${content}</body>`);
  win.document.close();

  win.setTimeout(() => {
    win.print();
    win.close();
  }, 0);
```

Reply:
I have just removed the ? after the win object, and checked it is created before working with it. @janibolkvadze, please check my comment and suggest a better option, if you have any. Thanks.

Reply:
Adding checking of `win` and removing the `?` looks good to me.